internal/cvdremote/cvdremote.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cvdremote
+
+import (
+	"flag"
+	"fmt"
+)
+
+type Command interface {
+	Run([]string) error
+}
+
+// Groups the subcommands handled in cvdremote.
+type CVDRemoteSubCommands struct {
+	CreateCVD   Command
+	ListCVDs    Command
+	ListHosts   Command
+	CreateHost  Command
+	DeleteHosts Command
+}
+
+const cvdremoteUsage = `NAME
+    cvdremote - manage Cuttlefish Virtual Devices (CVDs) in the cloud.
+
+SYNOPSIS
+    cvdremote [-h | -help] [<resource>] <command> [<args>]
+
+RESOURCES
+    cvd (default)
+        Cuttlefish Virtual Devices.
+
+    host
+        Host machines where CVDs live.
+
+COMMANDS
+    create
+        Create a resource.
+
+    list
+        List the resources.`
+
+type CVDRemoteCommand struct {
+	fs             *flag.FlagSet
+	subCommandsMap map[string]map[string]Command
+}
+
+func NewDefaultCVDRemoteCommand() Command {
```

Comment:
nit: The use of "Default" in the name implies that there are other valid `CVDRemoteCommand`s, which  is only true during testing. This function should be called `NewCVDRemoteCommand` and for testing you should call one called `NewTestingCVDRemoteCommand`.

Reply:
used `NewCVDRemoteCommand` and `NewCVDRemoteCommandWithArgs`